transport-native-unix-common/src/main/java/io/netty/channel/unix/Errors.java

```diff
@@ -102,6 +102,34 @@ public synchronized Throwable fillInStackTrace() {
         }
     }
 
+    public static final class NativeCallException extends RuntimeException {
+        private static final long serialVersionUID = -5532328671712318161L;
+        private final int expectedErr;
+        private final boolean fillInStackTrace;
+
+        public NativeCallException(String method, int expectedErr) {
+            this(method, expectedErr, true);
+        }
+
+        public NativeCallException(String method, int expectedErr, boolean fillInStackTrace) {
+            super(method + "(..) failed: " + errnoString(-expectedErr));
+            this.expectedErr = expectedErr;
+            this.fillInStackTrace = fillInStackTrace;
```

Comment:
Instead of this business with the `fillInStackTrace` field, you can pass it to the `writableStackTrace` parameter of `RuntimeException`.

See the protected constructor here: https://docs.oracle.com/javase/8/docs/api/java/lang/RuntimeException.html#RuntimeException-java.lang.String-java.lang.Throwable-boolean-boolean-

Reply:
The `NativeIoException` you copied above can't do this because `IOException` does not expose that constructor.

Reply:
Removed the class